New test analytics doc for docs.sentry (#14296)
getsentry/sentry-docs: docs/product/test-analytics/index.mdx

---
title: Test Analytics
sidebar_order: 135
description: Learn about how Sentry's Test Analytics tools can help you improve your code.
---

Sentry Test Analytics provides actionable insights into your CI test runs, helping you identify flaky tests, track failures, and optimize your test suite for faster, more reliable deployments. 

## Why Use Test Analytics?

- **Spot Flaky Tests:** Quickly identify and address tests that fail intermittently, reducing noise and wasted debugging time
- **Track Failures and Trends:** Monitor cumulative failures, slowest tests, and overall test run efficiency to improve CI reliability

> - **Actionable Insights:** See which lines of code failed within tests, confirm flaky tests, and prioritize fixes

Reviewer: ```suggestion
- **Get Actionable Insights:** See which lines of code failed within tests, confirm flaky tests, and prioritize fixes
```

> - **Integrated with Sentry:** Access test analytics alongside your error and performance data for a unified developer experience

Reviewer: ```suggestion
- **Leverage Sentry Integration:** Access test analytics alongside your error and performance data for a unified developer experience
```

My brain wanted all these little subheaders to agree a little more, feel free to push back :P

```

```bash {tabTitle:windows}
curl -L https://github.com/getsentry/sentry-prevent-cli/releases/latest/download/sentry-prevent-cli-windows.exe -o sentry-prevent-cli.exe
```

```bash {tabTitle:linux arm64}
curl -L https://github.com/getsentry/sentry-prevent-cli/releases/latest/download/sentry-prevent-cli-linux-arm64 -o sentry-prevent-cli
chmod +x sentry-prevent-cli
```

```bash {tabTitle:alpine linux arm64}
curl -L https://github.com/getsentry/sentry-prevent-cli/releases/latest/download/sentry-prevent-cli-alpine-arm64 -o sentry-prevent-cli
chmod +x sentry-prevent-cli
```

Then, upload your test results using the CLI:

```bash
sentry-prevent-cli upload --report-type test-results --token <PREVENT_TOKEN>
```

### Run Your Test Suite
Now that you've configured your CI to upload your test results, you can inspect the workflow logs to see if the call to Sentry succeeded. You need to have some failed tests to view the failed tests report. 

## Viewing Test Analytics

After your workflow runs, view failed tests in the [failed tests dashboard](https://sentry.io/prevent/failed-tests/), or in your GitHub pull request comments.

![Test Analytics comment](./img/TA-comment.png)

> You can see a complete overview of your test analytics in the [test analytics dashboard](sentry.io/prevent/).

Reviewer: ```suggestion
You can see a complete overview of your test analytics in the [test analytics dashboard](https://sentry.io/prevent/).
```

Without the protocol, markdown will interpret this as a relative path.

---
title: Test Analytics
sidebar_order: 135
description: Learn about how Sentry Prevent's Test Analytics tools can help you improve your code.
---

Sentry Prevent Test Analytics provides actionable insights into your CI test runs, helping you identify flaky tests, track failures, and optimize your test suite for faster, more reliable deployments.

## Why Use Test Analytics?

- **Spot Flaky Tests:** Quickly identify and address tests that fail intermittently, reducing noise and wasted debugging time.
- **Track Failures and Trends:** Monitor cumulative failures, slowest tests, and overall test run efficiency to improve CI reliability.
- **Actionable Insights:** See which lines of code failed within tests, confirm flaky tests, and prioritize fixes.

> - **Integrated with Sentry:** Access test analytics alongside your error and performance data for a unified developer experience. Yay

Author: Oh, goodness, this is related to the local env issue I'm having. That `yay` should have been deleted. 🤭
